### DISCUSSION TITLE
How do I make the RTSP work on my home network?

### DISCUSSION THREAD

**Discussion by SoyG** [tsightler/ring-mqtt#891] [Support Q&A]
I have the RTSP URL, which looks like "rtsp://000000-ring-mqtt:8554/0000000_live", but that isn't a valid URL. There's no ".com" or ".local" or any domain so it just comes up as an invalid URL.

**Comment by tsightler** [comment]
Please describe in more detail what you are attempting to do.  The URL is completely valid, domains are not a requirement for a URL, just a resolvable hostname, and that hostname should be resolvable on the internal Docker network (well, assuming you are using the correct hostname and not "000000") where the addon runs. 

If you are attempting to connect to the RTSP URL from a host other than the one on which the container is running, then you have to expose the RTSP port externally in the configuration and then replace the hostname portion of the URL with the name of the host/IP of the system on which the container is running.

**Reply by SoyG** [reply]
Yes the zeroes are because I thought I'm supposed to blank those out. So I'm using the Home Assistant Green. I can view a Ring video stream on the MQTT addon in the HA app. My goal is I'm trying to set it up to cast to a Google Assistant screen when the Ring detects motion. As far as I can tell, ring-mqtt is connected to MQTT because it shows them as devices with video and snapshots, but it's only showing the last event instead of a live stream.

But first, I'm trying just to view it externally. I tried replacing the hostname with the IP address of the Green and open it in VLC on my iPad (on the same network) in the format rtsp://192.168.1.122:8554/f4e11exxx_live but it just says video failed. If I try to open with the hostname instead of the IP, VLC just sits there with a black screen and then closes after a few seconds. I'm not sure if I exposed it externally. I'm also using HA cloud if that helps

**Reply by SoyG** [reply]
Wait I got it. I checked the box at the bottom of the addon and put port 8554. Looks like thats working now. 
How can I have it "cast" to my Google screen?

**Reply by tsightler** [reply]
That's more of a HA question that anything ring-mqtt specific, just a standard automation with play media action is all that is required.  Plenty of information out there already.

**Reply by SoyG** [reply]
Thanks I was able to get it working by using the Generic Camera device. It's too slow to start the video even with ethernet-joined cameras and displays, so it's not very practical and more of a fun gimmick. I'm guessing I can try the RTSP to webrtc addon, but how much does it actually improve?

**Reply by tsightler** [reply]
The difference with the webrtc addon is pretty significant.  HA uses HLS for it's stream engine and, even though it is the low-latency variant, it still requires 3-5 seconds of buffering on top of the typically 2-3 second startup time for the RTSP stream.  The WebRTC component eliminates nearly all of that.  The startup and response latency using the WebRTC addon is roughly equivalent to the Ring web dashboard + a few hundred milliseconds.

However, how much this impacts casting, I'm not sure, that adds yet a bunch more things that have to be negotiated and many cast devices are super under powered and optimized for playing the stream smooth vs startup time, thus they buffer quite a bit before starting.  I've seen some postings that indicate MJPEG streams start way faster on chromecast devices, and I do plan to introduce those at some point, but I've had that plan for a year and just haven't been able to find time to work on it.

Also, ethernet wired doesn't matter too much since everything has to go to Ring cloud (AWS) and back anyway, and that is far more latency than local wifi no matter what internet connection you have.

**Reply by SoyG** [reply]
Ok wow! You weren't kidding, the webrtc difference is night and day, it's nearly instant now. Also the install was far easier than I expected. I just installed the RtscToWebRTC addon, and it automatically found the RTSP cameras I configured with ring-mttq. 

So far, I've only figured out how to create a picture card, and an http link to the stream. I don't see how to send the stream to another device, because it doesn't appear as a new camera or device in HA. You mentioned Go2RTC? I grabbed that, but that one doesn't seem to automatically configure anything. Do I also need to get the "WebRTC Camera" addon to make it a camera? https://github.com/AlexxIT/WebRTC 

One other question: The RTSP to webrtc addon has an option in the web UI for "Persistent connection" or "On Demand". I'm assuming this should not be set to "Persistent" for a Ring camera, otherwise it'll just stay in live view forever? That might be why opening it was nearly instant

**Reply by tsightler** [reply]
I'm probably not going to be much help at this point as I don't do anything sophisticated with my setup.  I just use the go2rtc addon, which automatically discovers the HA cameras and makes them available as WebRTC, in concert with the RTSPtoWebRTC integration so that the native picture card uses WebRTC instead of HLS, which sounds like what you have done already.

However, I know that the WebRTC card also has some built-in support for casting so it might be required for that use case, I really don't use that feature at all as I have no application for it.
